api/src/main/java/javax/enterprise/inject/se/SeContainerInitializer.java

```diff
@@ -0,0 +1,317 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.se;
+
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import java.lang.annotation.Annotation;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+/**
+ * Container initializer to se CDI in Java SE.
+ * It is obtained by calling the {@link SeContainerInitializer#newInstance()} static method
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer container = SeContainerInitializer.newInstance().initialize();
+ * container.select(Foo.class).get();
+ * container.close();
+ * </pre>
+ * <p>
+ * <p>
+ * Since {@link SeContainer} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (SeContainer container = SeContainerInitializer.newInstance().initialize()) {
+ *     container.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer container = SeContainerInitializer.newInstance().addBeanClasses(Foo.class, Bar.class).addAlternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer container = SeContainerInitializer.newInstance().disableDiscovery().addBeanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainerInitializer containerInitializer = SeContainerInitializer.newInstance()
+ *    .disableDiscovery()
+ *    .addPackages(Main.class, Utils.class)
+ *    .addInterceptors(TransactionalInterceptor.class)
+ *    .addProperty("property", true);
+ * SeContainer container = container.initialize();
+ * </pre>
+ *
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ * @since 2.0
+ */
+public abstract class SeContainerInitializer {
+
+
+    /**
+     * Returns an instance of {@link SeContainerInitializer}
+     * Each call returns a new instance
+     *
+     * @return a new SeContainerInitializer instance.
+     * @throws IllegalStateException if called in a Java EE container
```

Comment:
Well, the SE package may probably be put in a specific jar so shouldn't be accessible in Java EE. But if a user put it in his deployment the serviceLoader won't be able to find an implementation for `SeContainerInitializer` and will throw this exception (see `findSeContainerInitializer()`)